docs/DeveloperGuide.md

```diff
@@ -196,33 +196,35 @@ The edit item feature allows the user to edit an `Item` currently being tracked
-component then saves the edited data into `data/trackO.json`.
+The edit item command `editi` is implemented through the `Logic` and `Model` components. The `Logic`
+component parses the user input, the `Model` component then performs the edit on the target `Item`.
 
-Step 1. The user inputs the command `editi 1 i/Chair q/20`. This calls `LogicCommand#execute()` which calls
-`TrackOParser#parseCommand`. This parses the command as an `EditItemCommand` and returns an `EditItemCommandParser`. 
-`EditItemCommandParser#parse()` parses the arguments and returns an `EditItemCommand` with the target index and the
-appropriate `EditItemDescriptor` as input. The `EditItemDescriptor` contains information which the newly edited `Item`
+Step 1: The user inputs the command `editi 1 i/Chair q/20`. This calls:
+1. `LogicManager#execute()`
+2. `TrackOParser#parseCommand()`. This parses the command as an `EditItemCommand` and returns an `EditItemCommandParser` object.
+3. `EditItemCommandParser#parse()` parses the arguments and returns an `EditItemCommand` with the target `Index` and the
+appropriate `EditItemDescriptor` as input.
+
+The `EditItemDescriptor` contains information which the newly edited `Item`
 should have and is used in the creation of the new `Item` object. In this case, the `EditItemDescriptor` contains a new
-`ItemName` and `Quantity` taken from the user input, while all other fields are copied from the existing item at the
-target index 1.
+`ItemName` and `Quantity` taken from the user input, while all other fields are copied from the existing `Item` at the
+target `Index` 1.
+
+Step 2:. The `EditItemCommand` creates a new `Item` using `createEditedItem()` and the `EditItemDescriptor`. It then
+checks if this `Item` already exists in the inventory list by using `ModelManager#hasItem()`. If it already exists, a
+`CommandException` is thrown with `MESSAGE_DUPLICATE_ITEM`.
 
-Step 2. The `EditItemCommand` creates a new `Item` using `createEditedItem()` and the `EditItemDescriptor`. It then
-checks if this `Item` already exists in the inventory list by using `Model#has()`. If it already exists, a
-`CommandException` is thrown with `MESSAGE_DUPLICATE_ITEM`. An item already exists if there is another item in the
-inventory list with same `ItemName`. `Item#isSameItem` returns true when both `Item` have the same `ItemName`. This is
+An item already exists if there is another item in the
+inventory list with same `ItemName`. `Item#isSameItem()` returns true when both `Item` have the same `ItemName`. This is
 because having 2 `Item` with the same `ItemName` can be confusing to the user and this safeguards the user from such a
 situation.
 
-Step 3. The `Item` at the target index is then replaced by the newly created `Item` using `Model#setItem()`,
+Step 3. The `Item` at the target index is then replaced by the newly created `Item` using `ModelManager#setItem()`,
```

Comment:
I think `Model#setItem()` should be fine actually, to match the sequence diagram!